src/SSHDebugPS/ConnectionManager.cs

```diff
@@ -20,54 +21,102 @@ internal static Connection GetInstance(string name, ConnectionReason reason)
-                    });
+                    userName = name.Substring(0, atSignIndex);
+                    hostName = name.Substring(atSignIndex + 1);
                 }
-                catch (Exception ex)
+                else
                 {
-                    VsShellUtilities.ShowMessageBox(ServiceProvider.GlobalProvider, ex.Message, null, 
-                        OLEMSGICON.OLEMSGICON_CRITICAL, OLEMSGBUTTON.OLEMSGBUTTON_OK, OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST);
-                    return null;
+                    userName = string.Format(CultureInfo.CurrentCulture, StringResources.UserName_PlaceHolder);
+                    hostName = name;
                 }
 
-                // NOTE: This will be null if connect is canceled
-                if (remoteSystem != null)
+                PasswordConnectionInfo newConnectionInfo = new PasswordConnectionInfo(hostName, userName, new System.Security.SecureString());
+
+                IConnectionManagerResult result = connectionManager.ShowDialog(newConnectionInfo);
+
+                if (result.DialogResult == ConnectionManagerDialogResult.Succeeded)
                 {
-                    return new Connection(remoteSystem);
+                    // Retrieve the newly added connection
+                    store.Load();
+                    connectionInfo = store.Connections.First(info => info.Id == result.StoredConnectionId);
                 }
             }
-            else
+
+            if (connectionInfo != null)
             {
-                // TODO: Replace this with credentials dialog from Linux extension
-                Connection connection = null;
+                remoteSystem = new UnixSystem();
 
-                try
+                while (true)
                 {
-                    connection = VS.CredentialsDialog.Show(name, reason);
-                    store.Add(connection.ConnectionInfo);
+                    try
+                    {
+                        VSOperationWaiter.Wait(string.Format(CultureInfo.CurrentCulture, StringResources.WaitingOp_Connecting, name), throwOnCancel: false, action: () =>
+                        {
+                            remoteSystem.Connect(connectionInfo);
+                        });
+                        break;
+                    }
+                    catch (RemoteAuthenticationException)
+                    {
+                        IVsConnectionManager connectionManager = (IVsConnectionManager)ServiceProvider.GlobalProvider.GetService(typeof(IVsConnectionManager));
+                        IConnectionManagerResult result = connectionManager.ShowDialog(StringResources.AuthenticationFailureHeader, StringResources.AuthenticationFailureDescription, connectionInfo);
+
+                        if (result.DialogResult == ConnectionManagerDialogResult.Succeeded)
+                        {
+                            // Update the credentials in the store
```

Comment:
This should probably be automatically handled by the dialog. Will change after dialog makes that change.